### PR TITLE
[FEATURE] refactoring of coarse isotope distributions

### PR DESCRIPTION
**Pull request by timosachsenberg**
This PR only contains the changes induced by the refactoring. 
Same functionality as before is provided but preparations are made to choose an isotope generator algorithm at run time. It is a subset of https://github.com/OpenMS/OpenMS/pull/3173

### CODE REVIEW

**Comment by timosachsenberg** [comment]
TODO:
- adjust documentation (check all members that changed)
- adjust pyOpenms files (pxd files that need to be adapted are located in the : src/pyOpenMS/pxds/ folder. See e.g.: EmpiricalFormula.pxd)

File: src/openms/include/OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/Container.h [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow $
+// $Authors: Clemens Groepl, Andreas Bertsch, Chris Bielow $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_CHEMISTRY_ISOTOPEDISTRIBUTION_CONTAINER_H
+#define OPENMS_CHEMISTRY_ISOTOPEDISTRIBUTION_CONTAINER_H
```
Comment (hroest) [inline src/openms/include/OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/Container.h]: why is the file called "Container.h" ? seems a bit generic

**Comment by hroest** [comment]
can fix pyOpenMS and the style tests http://cdash.openms.de/viewTest.php?onlyfailed&buildid=254480

File: src/openms/include/OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/Container.h (hunk 1 above)
Comment (timosachsenberg) [reply src/openms/include/OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/Container.h]: In the class it is still called IsotopeDistribution.
I will rename the file accordingly...

**Comment by hroest** [comment]
@timosachsenberg what is the status here, do you still want to go forward with this?

**Comment by timosachsenberg** [comment]
Nikos will join the developer meeting and then have some time to fix the remaining issues.

**Comment by pati-ni** [comment]
@hroest I am on it, I have resolved the conflicts with the development branch a few days ago. I am getting some difficulties with limited time, slow laptop and building successfully pyOpenMS. I am close getting to a solution that won't contain duplicated code and will replicate the interface hierarchy.

**Pull request closed without merging**